src/router/RouterApp.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext } from "react";
+import { Routes, Route, Navigate } from "react-router-dom";
+import { TokenContext } from "../context/TokenContext";
+
+const RouterApp = () => {
+  const { user } = useContext(TokenContext);
+  return (
+    <Routes>
+      {user ? (
```

Comment:
no way
build a protecred Route component with dynamic guards
I do want to see guards array prop, to pass multiple guards to the protected route component

example usage:
```
<ProtectedRoute path="employees" exact guards={[authGuard]} element={<EmployeesPage />} />
<ProtectedRoute path="languages" exact guards={[authGuard, roleGuard('admin')]} element={<LanguagesPage/>}/>
<Route path="login" exact element={<LoginPage />} />
```

if fail authGuard - redirect to login
if fail roleGuard - just dont show page, no redirects, OR redirect to 403/404